java/plasma/src/main/java/org/apache/arrow/plasma/exceptions/PlasmaOutOfMemoryException.java

```diff
@@ -22,11 +22,11 @@
  */
 public class PlasmaOutOfMemoryException extends RuntimeException {
 
-  public PlasmaOutOfMemoryException() {
+  public PlasmaOutOfMemoryException(String message) {
```

Comment:
Shouldn't we pass the string somewhere? eg add it to the original message?

Reply:
In plasma/lib/java/org_apache_arrow_plasma_PlasmaClientJNI.cc
Line 123
`  env->ThrowNew(exceptionClass, "");`
Here passed an empty String.But There is no String arg in `PlasmaOutOfMemoryException ` constructor.
And if we need to catch this exception, it would show `java (Ljava/lang/String;Ljava/lang/String;I)V`
So add a String arg in PlasmaOutOfMemoryException constructors.

Reply:
An unit test added, please take a review.
@rymurr

Reply:
thanks for the clarity @offthewall123 

I think it would be better to add new constructors to augment the existing one and to append/replace the `String message` to the internal message.

Reply:
Thanks for reply. @rymurr 
Yes, we can add a new constructor like `public PlasmaOutOfMemoryException(String message) ` and keep the origin two.
But In plasma/lib/java/org_apache_arrow_plasma_PlasmaClientJNI.cc
line 123 `env->ThrowNew(exceptionClass, "");` here just passed an empty String, and i searched seems no overloaded function like `env->ThrowNew(exceptionClass)` which not to pass a String arg. 

If we replace `String message` to the internal message -> It's an empty String, i think we should keep the internal message `super("The plasma store ran out of memory.");`

If Add new constructors -> The origin two constructors are of no use.

So `String message` added here is just to adapt `env->ThrowNew(exceptionClass, "");`

Reply:
Hey @offthewall123 My concern is that someone somewhere might be using those exceptions in other ways and we shouldn't break their code. I think something like `super("The plasma store ran out of memory. " + message)` will result in effectively no change from current behaviour but if someone does add more context to the exception then the user will know about it.

Reply:
Hi @rymurr, i got it. Added a new constuctor without breaking the origin two.Unit test passed also.

Reply:
@rymurr Could take a review again?

Reply:
@wesm